src/scenes/Engagement/InternshipEngagement/InternshipEngagementDetail.tsx

```diff
@@ -64,156 +75,228 @@ export const InternshipEngagementDetail: FC<EngagementQuery> = ({
               <Grid item xs={6}>
-                {node}
+                <FieldOverviewCard
+                  title="Growth Plan Complete Date"
+                  data={{
+                    value: formatDate(engagement.completeDate.value),
+                  }}
+                  icon={PlantIcon}
+                  onClick={() => show('completeDate')}
+                  onButtonClick={() => show('completeDate')}
+                />
+              </Grid>
+              <Grid item xs={6}>
+                <FieldOverviewCard
+                  title="Disbursement Complete Date"
+                  data={{
+                    value: formatDate(
+                      engagement.disbursementCompleteDate.value
+                    ),
+                  }}
+                  icon={OptionsIcon}
+                  onClick={() => show('disbursementCompleteDate')}
+                  onButtonClick={() => show('disbursementCompleteDate')}
+                />
               </Grid>
-            )}
-            onEdit={() => show('mentorId')}
-          />
+              <Grid item xs={6}>
+                <FieldOverviewCard
+                  title="Communications Complete Date"
+                  data={{
+                    value: formatDate(
+                      engagement.communicationsCompleteDate.value
+                    ),
+                  }}
+                  icon={ChatOutlined}
+                  onClick={() => show('communicationsCompleteDate')}
+                  onButtonClick={() => show('communicationsCompleteDate')}
+                />
+              </Grid>
+              <Grid item container spacing={3} alignItems="center">
+                <Grid item xs={6}>
+                  <Typography variant="h4">Growth Plan</Typography>
+                </Grid>
+              </Grid>
+              <Grid item container spacing={3} alignItems="center">
+                <Grid item xs={6}>
+                  {growthPlan.canRead && !growthPlan.value ? (
+                    <AddItemCard
+                      actionType="dropzone"
+                      canAdd={growthPlan.canEdit}
+                      DropzoneProps={{
+                        classes: { text: classes.dropzoneText },
+                      }}
+                      handleFileSelect={(files: File[]) =>
+                        uploadFile({ files, parentId: engagement.id })
+                      }
+                      itemType="Growth Plan"
+                    />
+                  ) : (
+                    <DefinedFileCard
+                      onVersionUpload={(files) =>
+                        uploadFile({
+                          action: 'version',
+                          files,
+                          parentId: engagement.id,
+                        })
+                      }
+                      resourceType="engagement"
+                      securedFile={engagement.growthPlan}
+                    />
+                  )}
```

Comment:
Maybe it would be better to split this out into a separate GrowthPlan component? There's a lot going on in this file. I think we could move the useEngagementFile hook there and then only wrap this section with the context instead of the whole component. Thoughts?

Reply:
There's a lot going on, but not much of it is dedicated to handling the logic of whether it's being used for a Growth Plan vs. a PNP. I'm not sure we'd gain a lot by splitting it up. But I think we can wrap just the card with the context without doing that. I don't think `useUploadEngagementFile` hook needs to be inside the context; it just needs to be inside the `UploadContext`

Reply:
Okay, I made this change, and it worked just fine

Reply:
> There's a lot going on in this file.

Sorry, I just realized you probably meant "the `InternshipEngagementDetail` file, not the `DefinedFileCard` file. In which case, yes, but no more so than in some other scenes that have conditional display of components. All the other "detail" scene files have all their UI elements right there in the file... it seems like kind of a shame to break that pattern when it's so predictable. It's kind of nice not to have to click around through multiple layers to work on some part of a detail scene